api/include/opentelemetry/trace/span.h

```diff
@@ -99,6 +101,94 @@ class Span
                        attributes.begin(), attributes.end()});
   }
 
+#if OPENTELEMETRY_ABI_VERSION_NO >= 2
+
+  /**
+   * Add link (ABI).
+   *
+   * @since ABI_VERSION 2
+   */
```

Comment:
as suggested in the [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#link), should we mention this in document

"The API documentation MUST state that adding links at span creation is preferred to calling AddLink later, for contexts that are available during span creation, because head sampling decisions can only consider information present during span creation."

Reply:
Thanks, and fixed in doxygen doc.

I added comments for attributes while at it.